pkgs/development/compilers/fasmg/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv
+, fetchzip
+}:
+
+stdenv.mkDerivation rec {
+  pname = "fasmg";
+  version = "j27m";
+
+  src = fetchzip {
+    url = "https://flatassembler.net/fasmg.${version}.zip";
+    sha256 = "0qmklb24n3r0my2risid8r61pi88gqrvm1c0xvyd0bp1ans6d7zd";
+    stripRoot = false;
+  };
+
+  buildPhase = with stdenv; let
+    if64 = lib.optionalString isx86_64;
+    fasmg-bin =
```

Comment:
It would be a bit more clear:
```nix
  inherit (stdenv.hostPlatform) system;
  throwSystem = throw "Unsupported system: ${system}";

  fasm-bin = {
    x86_64-linux = "fasm.x64";
    x86_64-darwin = "source/macos/x64/fasm";
    x86-linux = "fasm";
    x86-darwin = "source/macos/fasm";
  }.${system} or throwSystem;